src/lib/pages/backoffice/Loan/LoanDetails/LoanActions/LoanActions.js

```diff
@@ -1,16 +1,53 @@
 import { OverdueLoanSendNotificationModal } from '@modules/Loan/backoffice/OverdueLoanSendNotificationModal';
 import { InfoMessage } from '@components/backoffice/InfoMessage';
+import { withCancel } from '@api/utils';
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { documentApi } from '@api/documents';
 import { List, Button } from 'semantic-ui-react';
+import { RenderExtend } from './ExtendButton';
+import { RenderCancel } from './CancelButton';
 import { omit } from 'lodash/object';
-import { CancelModal } from '@components/CancelModal';
 import _isEmpty from 'lodash/isEmpty';
 import capitalize from 'lodash/capitalize';
 
 export default class LoanActions extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      overbooked: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchCirculation();
+  }
+
+  componentWillUnmount() {
+    this.cancellableCirculation && this.cancellableCirculation.cancel();
+  }
+
+  fetchCirculation = async () => {
```

Comment:
this function should be moved to RenderExtend component, to separate the concern of checking if the document is overbooked